scipy/stats/_stats_py.py

```diff
@@ -2821,26 +2821,35 @@ def obrientransform(*samples):
     that the variances are different.
 
     """
-    TINY = np.sqrt(np.finfo(float).eps)
+    xp = array_namespace(*samples)
 
     # `arrays` will hold the transformed arguments.
     arrays = []
     sLast = None
+    dtype = None
 
     for sample in samples:
-        a = np.asarray(sample)
-        n = len(a)
```

Comment:
Hmm. I thought that the behavior of this function was like `axis=None` before because the `mean` and `sum` functions are used without an explicit `axis` argument, and I think I saw `xp_size` below. I didn't notice this `n=len(a)`. This means that the function was simply *wrong* for n-d arrays before; it did not produce a valid result equivalent to `axis=None`. So I don't know if an explicit deprecation of leaving `axis` unspecified is really required to `axis` with a default of `0`. : /

We *could* still add it and require that it be specified explicitly, but I think it would just be an excuse for changing the output type, not for adding `axis` with a default of `0`.